lib/matplotlib/axes/_axes.py

```diff
@@ -2391,12 +2391,12 @@ def bar(self, x, height, width=0.8, bottom=None, *, align="center",
 
         if orientation == 'vertical':
             self._process_unit_info(
-                [("x", x), ("y", height)], kwargs, convert=False)
+                [("x", x), ("y", height + y)], kwargs, convert=False)
```

Comment:
Should we use `_convert_dx` for these?

Reply:
Grrrr... our units handling is such a PITA.  The problem is `height = [1, 2, 4]`, and `y=bottom = 0` of course cannot be added together.  `y=[0]` won't help that, and indeed is wildly incorrect.  

Trying by extracting the first safe element from x and y.  This should maybe move into the units checker...